tests/foreman/ui_airgun/test_puppetenvironment.py

```diff
@@ -0,0 +1,61 @@
+from pytest import raises
+from nailgun import entities
+import pytest
+from robottelo.datafactory import (
+    gen_string,
+    invalid_values_list,
+)
+
+
+@pytest.fixture(scope='module')
+def init_values():
+    """Fixture returns values for new environment"""
+    name = gen_string('alpha')
+    org = entities.Organization().create()
+    location = entities.Location().create()
+    puppetEnvironmentValues = {
+        'environment.name': name,
+        'locations.resources.assigned': [location.name],
+        'organizations.resources.assigned': [org.name],
+    }
+    return puppetEnvironmentValues
+
+
+def test_positive_create(session, init_values):
+    name = init_values.get('environment.name')
+    with session:
+        session.puppetenvironment.create(values=init_values)
+        assert session.puppetenvironment.search(
+            name)[0]['Name'] == name
+        env_values = session.puppetenvironment.read(name)
+        assert env_values['locations']['resources']['assigned'][0] == \
+            init_values.get('locations.resources.assigned')[0]
+
+
+def test_negative_create(session):
+    for name in invalid_values_list(interface='ui'):
+        with session:
+            session.puppetenvironment.create({'name': name})
+            with raises(AssertionError):
+                assert session.puppetenvironment.search(
```

Comment:
Ideally your helpers should verify notification messages after creation so `AssertionError` should be risen during `create` call.
This `raises()` for search is redundant, just don't compare with expected name and make sure there's no results instead:
```python
assert not session.puppetenvironment.search(name)
```